### ISSUE TITLE
Using AddElement in meta data editor leads to Exception (normal meta data editor)

### ISSUE THREAD

**Issue by stefanCCS** [kitodo/kitodo-production#3820]
Version 3.2-build 16.09.2020
I was able to produce an exeption doing the following:
- created 3 "chapter" 
- moving 1st page to 1.Chapter --> ok
- moving last page to 3. chapter --> Exception

See also #3819 
![addElement-B-1](https://user-images.githubusercontent.com/61045115/86929258-f10fe680-c135-11ea-90db-a341a1a6b08f.png)

**Comment by solth** [comment]
@stefanCCS is that the same `IllegalArgumentException` as in #3819 ?

**Comment by stefanCCS** [comment]
If a select a page, after adding first element (which works fine), I get the following exception:
n error has occurred:
Date/time: 2020-07-09 12:45:47
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36
User IP: 192.168.47.2
Request URI: /kitodo-3.2.0-20200619/pages/metadataEditor.jsf
Ajax request: Yes
Status code: 500
Exception type: class java.lang.IllegalArgumentException
Exception message: java.lang.ClassCastException@704de550
Exception UUID:
Stack trace:
java.lang.IllegalArgumentException: java.lang.ClassCastException@704de550
	at jdk.internal.reflect.GeneratedMethodAccessor1441.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.kitodo.production.forms.createprocess.ProcessFieldedMetadata.preserve(ProcessFieldedMetadata.java:515)
	at org.kitodo.production.forms.dataeditor.MetadataPanel.preserveLogical(MetadataPanel.java:152)
	at org.kitodo.production.forms.dataeditor.DataEditorForm.switchStructure(DataEditorForm.java:596)
	at org.kitodo.production.forms.dataeditor.StructurePanel.treeLogicalSelect(StructurePanel.java:703)
	at jdk.internal.reflect.GeneratedMethodAccessor1451.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.el.parser.AstValue.invoke(AstValue.java:247)
	at org.apache.el.MethodExpressionImpl.invoke(MethodExpressionImpl.java:267)
	at org.jboss.weld.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40)
	at org.jboss.weld.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50)
	at org.apache.myfaces.view.facelets.el.ContextAwareTagMethodExpression.invoke(ContextAwareTagMethodExpression.java:96)
	at org.primefaces.behavior.ajax.AjaxBehaviorListenerImpl.processArgListener(AjaxBehaviorListenerImpl.java:84)
	at org.primefaces.behavior.ajax.AjaxBehaviorListenerImpl.processAjaxBehavior(AjaxBehaviorListenerImpl.java:74)
	at org.primefaces.event.AbstractAjaxBehaviorEvent.processListener(AbstractAjaxBehaviorEvent.java:48)
	at javax.faces.component.behavior.BehaviorBase.broadcast(BehaviorBase.java:71)
	at javax.faces.component.UIComponentBase.broadcast(UIComponentBase.java:416)
	at org.primefaces.component.api.UITree.broadcast(UITree.java:377)
	at org.primefaces.component.api.UITree.broadcast(UITree.java:397)
	at javax.faces.component.UIViewRoot._broadcastAll(UIViewRoot.java:1255)
	at javax.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:420)
	at javax.faces.component.UIViewRoot._process(UIViewRoot.java:1741)
	at javax.faces.component.UIViewRoot.processApplication(UIViewRoot.java:935)
	at org.apache.myfaces.lifecycle.InvokeApplicationExecutor.execute(InvokeApplicationExecutor.java:42)
	at org.apache.myfaces.lifecycle.LifecycleImpl.executePhase(LifecycleImpl.java:195)
	at org.apache.myfaces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:142)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:204)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.kitodo.production.servletfilter.EncodingFilter.doFilter(EncodingFilter.java:69)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:317)
	at org.kitodo.production.security.SecurityObjectAccessFilter.doFilter(SecurityObjectAccessFilter.java:86)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:127)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:91)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:114)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:137)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:111)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:170)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.session.ConcurrentSessionFilter.doFilter(ConcurrentSessionFilter.java:155)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:200)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:116)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:66)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:105)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:56)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:107)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:331)
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:214)
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:177)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:347)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:263)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.logging.log4j.web.Log4jServletFilter.doFilter(Log4jServletFilter.java:71)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:200)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:137)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:660)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:798)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:806)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1498)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:834)

**Comment by stefanCCS** [comment]
> 
> 
> @stefanCCS is that the same `IllegalArgumentException` as in #3819 ?

At least the behaviour is different:
In #3819 I could not even create the first element!

**Comment by solth** [comment]
This may be related to or even a duplicate of #3444

**Comment by stefanCCS** [comment]
When I get a newer version, I will test this again.

**Comment by stefanCCS** [comment]
Tested again with version
3.2-build-14.08.2020
Same behaviour with the following exception:
![AddElementException-3820](https://user-images.githubusercontent.com/61045115/90236042-80d63e00-de22-11ea-9405-950b2f9e03f0.png)

**Comment by matthias-ronge** [comment]
The error can still occur if an attempt is made to set an element that does not exist. At the moment only [`label`](https://github.com/kitodo/kitodo-production/blob/bfe5a55868d1e0c6510b2271c01f6f3590cf57d6/Kitodo-API/src/main/java/org/kitodo/api/dataformat/IncludedStructuralElement.java#L35) and [`orderlabel`](https://github.com/kitodo/kitodo-production/blob/bfe5a55868d1e0c6510b2271c01f6f3590cf57d6/Kitodo-API/src/main/java/org/kitodo/api/dataformat/IncludedStructuralElement.java#L58) exist at logical structures, and only [`orderlabel`](https://github.com/kitodo/kitodo-production/blob/bfe5a55868d1e0c6510b2271c01f6f3590cf57d6/Kitodo-API/src/main/java/org/kitodo/api/dataformat/MediaUnit.java#L62) and [`metsDivReferrerId`](https://github.com/kitodo/kitodo-production/blob/bfe5a55868d1e0c6510b2271c01f6f3590cf57d6/Kitodo-API/src/main/java/org/kitodo/api/dataformat/MediaUnit.java#L72) at physical structures. All other attempts will result in the error.

**Comment by Kathrin-Huber** [comment]
fixed by #3944

**Comment by stefanCCS** [comment]
Confirmed - works for me in official Release 3.2
